modules/base/rendering/renderabletrail.cpp

```diff
@@ -448,31 +479,81 @@ void RenderableTrail::render(const RenderData& data, RendererTasks&) {
     //    global::renderEngine->openglStateCache().resetDepthState();
     //    return;
     //}
+    
+    if (_splitTrailRenderMode) {
+        // Splits the trail up into three parts for more accurate rendering
+        // of renderableTrailTrajectory trails
 
-    // Render the primary batch of vertices
-    internalRender(
-        renderLines,
-        renderPoints,
-        data,
-        modelTransform,
-        _primaryRenderInformation,
-        totalNumber,
-        primaryOffset
-    );
-
-    // The secondary batch is optional, so we need to check whether we have any data here
-    if (_floatingRenderInformation._vaoID != 0 && _floatingRenderInformation.count != 0) {
+        internalRender(
+            renderLines,
+            renderPoints,
+            data,
+            modelTransform,
+            _primaryRenderInformation,
+            _primaryRenderInformation.count,
+            _primaryRenderInformation.first,
+            _splitTrailRenderMode,
+            _numberOfUniqueVertices
+        );
+        
+        int floatingOffset = std::max(0, _primaryRenderInformation.count - 1);
         internalRender(
             renderLines,
             renderPoints,
             data,
             modelTransform,
             _floatingRenderInformation,
+            _floatingRenderInformation.count,
+            _floatingRenderInformation.first,
+            _splitTrailRenderMode,
+            _numberOfUniqueVertices,
+            floatingOffset
+        );
+
+        int offset = (_floatingRenderInformation.count > 0) ? 1 : 0;
+        internalRender(
+            renderLines,
+            renderPoints,
+            data,
+            modelTransform,
+            _secondaryRenderInformation,
+            _secondaryRenderInformation.count,
+            _secondaryRenderInformation.first,
+            _splitTrailRenderMode,
+            _numberOfUniqueVertices,
+            offset
+        );
+        
+    }
+    else {
+        // Render the primary batch of vertices
+        internalRender(
+            renderLines,
+            renderPoints,
+            data,
+            modelTransform,
+            _primaryRenderInformation,
             totalNumber,
-            // -1 because we duplicate the penultimate point between the vertices
-            -(primaryOffset + _primaryRenderInformation.count - 1)
+            primaryOffset
         );
+
+        // The secondary batch is optional, so we need to check whether we have any data here
+        if (_floatingRenderInformation._vaoID != 0 && _floatingRenderInformation.count != 0) {
+            internalRender(
```

Comment:
Not sure that I completely follow.
The `offset` variable is not needed here as it's not doing any split rendering.

Reply:
Sorry, mb.

Higher up you defined:
```
int offset = (_floatingRenderInformation.count > 0) ? 1 : 0;
```

I was saying that it could be more readable to:
```
const int offset = -(primaryOffset + _primaryRenderInformation.count - 1);
```
and then use `offset` in the function call instead.  But super tiny thing

Reply:
Sorry, not seeing how this is equivalent. Is it OK to skip this for now and check it later?